tests/test_edgeql_expressions.py

```diff
@@ -834,6 +834,27 @@ class TestExpressions(tb.QueryTestCase):
                     [mod],
                 )
 
+    async def test_edgeql_expr_variables_01(self):
+        await self.assert_query_result(
+            r'''SELECT (INTROSPECT TYPEOF <int64>$0).name;''',
+            {'std::int64'},
+            variables=(7,),
+        )
+
+    async def test_edgeql_expr_variables_02(self):
+        await self.assert_query_result(
+            r'''SELECT <str>$1 ++ (INTROSPECT TYPEOF <int64>$0).name;''',
```

Comment:
+ @elprans 

I'm not sure it's supposed to work like this. I think one should be able to have a query in `INTROSPECT TYPEOF` with a `$0` parameter and then use `$0` in their outer query, as in:

```
SELECT <str>$0 ++ (INTROSPECT TYPEOF <int64>$0).name;
```

That allows one to insert any query into `INTROSPECT TYPEOF` kind of construct without worrying about potential name/positional parameters collision.

Reply:
Disregard this comment; Elvis convinced me that the current approach is fine.

+1 to inlining unused SQL parameters in a CTE.